### DISCUSSION TITLE
`between_time(index)` has no method for class character

### DISCUSSION THREAD

**Discussion by jablko** [microsoft/finnts#160] [Q&A]
I'm just getting started with finnts (CRAN version, 0.4.0) and I'm getting the following error:
```
! `between_time(index)` has no method for class character
```

Is it obvious to anyone more experienced what I'm doing wrong?

Here are commands and input data I confirmed reproduce the error:
```sh
R -e '
  library(finnts)
  history <- read.csv("history.csv", colClasses = c(Date = "Date"))
  forecast_time_series(input_data = history, combo_variables = "Category", target_variable = "Amount", date_type = "month", forecast_horizon = 12)
'
```
```csv
"Category","Date","Amount"
"Groceries",22-11-01,-10
"Groceries",22-12-01,-177.39
"Groceries",23-01-01,-828.93
"Groceries",23-02-01,-299.05
"Groceries",23-03-01,-741.61
"Groceries",23-04-01,-1231.04
"Groceries",23-05-01,-1416.48
"Groceries",23-06-01,-344.27
"Groceries",23-07-01,-429.51
"Groceries",23-08-01,-1499.06
"Groceries",23-09-01,-494.03
"Groceries",23-10-01,-481.05
"Groceries",23-11-01,-1068.65
"Groceries",23-12-01,-777.18
"Groceries",24-01-01,-1489.97
"Groceries",24-02-01,-580.95
"Groceries",24-03-01,-1319.51
"Groceries",24-04-01,-583.98
"Groceries",24-05-01,-1074
```

and here's the complete session and backtrace:
```
$ R -e '
  library(finnts)
  history <- read.csv("history.csv", colClasses = c(Date = "Date"))
  forecast_time_series(input_data = history, combo_variables = "Category", target_variable = "Amount", date_type = "month", forecast_horizon = 12)
'

R version 4.2.2 Patched (2022-11-10 r83330) -- "Innocent and Trusting"
Copyright (C) 2022 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

>
>   library(finnts)
Loading required package: modeltime
>   history <- read.csv("history.csv", colClasses = c(Date = "Date"))
>   forecast_time_series(input_data = history, combo_variables = "Category", target_variable = "Amount", date_type = "month", forecast_horizon = 12)
Finn Submission Info
• Experiment Name: finn_fcst
• Run Name: finn_fcst-20240513T002850Z

✔ Prepping Data [12s]
✔ Creating Model Workflows [2.2s]
✔ Creating Model Hyperparameters [4.9s]
Registered S3 methods overwritten by 'readr':
  method                    from
  as.data.frame.spec_tbl_df vroom
  as_tibble.spec_tbl_df     vroom
  format.col_spec           vroom
  print.col_spec            vroom
  print.collector           vroom
  print.date_names          vroom
  print.locale              vroom
  str.col_spec              vroom
Error in `dplyr::filter()`:
ℹ In argument: `Date %>% between_time("start", end)`.
Caused by error:
! `between_time(index)` has no method for class character
Backtrace:
     ▆
  1. ├─finnts::forecast_time_series(...)
  2. │ └─finnts::prep_models(...)
  3. │   └─finnts:::train_test_split(...)
  4. │     └─... %>% ...
  5. ├─tidyr::separate(., col = .id, into = c(NA, "Slice_ID"), sep = "Slice")
  6. ├─timetk::tk_time_series_cv_plan(.)
  7. ├─timetk::time_series_cv(...)
  8. │ └─timetk:::period_chr_to_n(index_table_min, !!date_var_expr, period = initial)
  9. │   └─... %>% nrow()
 10. ├─base::nrow(.)
 11. ├─timetk::filter_by_time(...)
 12. ├─timetk:::filter_by_time.data.frame(...)
 13. │ └─.data %>% ...
 14. ├─dplyr::filter(...)
 15. ├─dplyr:::filter.data.frame(...)
 16. │ └─dplyr:::filter_rows(.data, dots, by)
 17. │   └─dplyr:::filter_eval(...)
 18. │     ├─base::withCallingHandlers(...)
 19. │     └─mask$eval_all_filter(dots, env_filter)
 20. │       └─dplyr (local) eval()
 21. ├─Date %>% between_time("start", end)
 22. ├─timetk::between_time(., "start", end)
 23. ├─timetk:::between_time.default(., "start", end)
 24. │ └─base::stop(...)
 25. └─base::.handleSimpleError(...)
 26.   └─dplyr (local) h(simpleError(msg, call))
 27.     └─rlang::abort(message, class = error_class, parent = parent, call = eror_call)
There were 37 warnings (use warnings() to see them)
✖ Creating Train Test Splits [1.3s]
Execution halted

**Comment by mitokic** [comment]
This is most likely due to not enough historical data. For a monthly forecast, especially one with a long forecast horizon, please try to get at least 3 years of historical data. 

If that's not possible. Then please adjust the back testing using the back_test_scenarios argument when calling Finn. Set it to something like 6. So Finn will then just back test on the last 6 months of your data. That should hopefully fix things.
